discovery-infra/node_controlers/node_controler.py

```diff
@@ -0,0 +1,132 @@
+class NodeControler(object):
```

Comment:
The docstrings for all these functions don't provide any extra info.
Use meaningful function names, parameters, and use type hints.
e.g.
Replace
```python
    def shutdown_node(self, node_name):
        """
        Shuts down node
        Parameters:
        node_name (str): name of node to shutdown
        Returns:
        None
        """
```
With
```python
    def shutdown_node(self, node_name: str) -> None:
       raise NotImplementedError
```

Reply:
done